### PR TITLE
Upgrade to ClosedXML 0.100.3

### PR DESCRIPTION
**Pull request by myo-min-tin**
Upgrade to ClosedXML v 0.100.3

### CODE REVIEW

File: ClosedXML.Report/Utils/IXLCellExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
+﻿using ClosedXML.Excel;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ClosedXML.Report.Utils;
```
Comment (vbondarev) [inline ClosedXML.Report/Utils/IXLCellExtensions.cs]: Feature 'file-scoped namespaces' is not available. Please use language version 10.0 or greater.

Comment (vbondarev) [reply ClosedXML.Report/Utils/IXLCellExtensions.cs]: Current version is 7

File: ClosedXML.Report/Excel/TempSheetBuffer.cs [hunk 2]
```diff
@@ -64,6 +66,27 @@ public IXLCell WriteValue(object value, IXLCell settingCell)
             return xlCell;
         }
 
+        public IXLCell WriteCellValue(XLCellValue value, IXLCell settingCell)
```
Comment (vbondarev) [inline ClosedXML.Report/Excel/TempSheetBuffer.cs:69]: Method 'WriteCellValue' return value is never used.

File: ClosedXML.Report/Excel/Subtotal.cs [hunk 3]
```diff
@@ -237,7 +237,7 @@ private void MoveSummary(MoveData moveData)
                 return;
 
             trgtRng.Clear();
-            fcell.Value = srcRng;
+            fcell.CopyFrom(srcRng); //fcell.Value = srcRng;
```
Comment (vbondarev) [inline ClosedXML.Report/Excel/Subtotal.cs]: Please remove all your commented text. We have git )

File: ClosedXML.Report/Utils/IXLCellExtensions.cs (hunk 1 above)
Comment (Ulisse67) [reply ClosedXML.Report/Utils/IXLCellExtensions.cs]: Using latest for LangVersion should use the highest compatible C# for the targeted framework, no need to set to an explicit version number.

Comment (vbondarev) [reply ClosedXML.Report/Utils/IXLCellExtensions.cs]: One of the supported frameworks is net462 which doesn't support c# 11.  Change language version to the `Latest` and try to compile.

Comment (Ulisse67) [reply ClosedXML.Report/Utils/IXLCellExtensions.cs]: I have tried with latest, and it builds, even on net462: of course using language features not supported by net462 is gonna fail whatever the language version specified, so I would try to keep it as wide as possible by using latest.

Comment (vbondarev) [reply ClosedXML.Report/Utils/IXLCellExtensions.cs]: I was wrong. If I set the Latest version is OK, but if I set the version is 11 I get the compile error   `Invalid option '11' for /langversion. Use '/langversion:?' to list supported values.`

Comment (Ulisse67) [reply ClosedXML.Report/Utils/IXLCellExtensions.cs]: csc /langversion:? by me is listing 11 as the latest (and default version, so I think now <langVersion> can be omitted to get the latest).
csc /version gives 4.5.0-6.23128.14 (651d32d8)

Comment (myo-min-tin) [reply ClosedXML.Report/Utils/IXLCellExtensions.cs]: Sorry, I don't have .net 4.62 on my development machine.
Please wrap IXLCellExtensions class into namespace ClosedXML.Report.Utils.

File: ClosedXML.Report/Excel/TempSheetBuffer.cs (hunk 2 above)
Comment (myo-min-tin) [reply ClosedXML.Report/Excel/TempSheetBuffer.cs:69]: Yes, 'WriteCellValue' return value is never used now. It's method overloading as WriteValue(object value, IXLCell settingCell).

File: ClosedXML.Report/Excel/Subtotal.cs (hunk 3 above)
Comment (myo-min-tin) [reply ClosedXML.Report/Excel/Subtotal.cs]: I removed my comments and then pushed it again.

Comment (vbondarev) [reply ClosedXML.Report/Excel/Subtotal.cs]: I cloned the branch and got the next error: 

```
 ClosedXML.Report.Tests.csproj: [NU1605] Warning As Error: Detected package downgrade: System.Linq.Dynamic.Core from 1.3.1 to 1.2.20. Reference the package directly from the project to select a different version. 
 ClosedXML.Report.Tests -> ClosedXML.Report -> System.Linq.Dynamic.Core (>= 1.3.1) 
 ClosedXML.Report.Tests -> System.Linq.Dynamic.Core (>= 1.2.20)
```

Comment (vbondarev) [reply ClosedXML.Report/Excel/Subtotal.cs]: Fixed error and run all tests:

image

Comment (Ulisse67) [reply ClosedXML.Report/Excel/Subtotal.cs]: Cannot say for all tests, but for this the test should be fixed:
image

It has to do with Cell.Value now being an XLCellValue type where before was object.
Same here:
image

**Comment by Pankraty** [comment]
Closed via #307.

@mymintin thank you for the try! Unfortunately, this particular upgrade turned out to be way harder than a usual "bump all dependencies up".

**Pull request closed without merging**